model/lifecycle.go

```diff
@@ -57,25 +57,81 @@ type VersionToRestart struct {
 // SetVersionActivation updates the "active" state of all builds and tasks associated with a
 // version to the given setting. It also updates the task cache for all builds affected.
 func SetVersionActivation(versionId string, active bool, caller string) error {
-	builds, err := build.Find(
-		build.ByVersion(versionId).WithFields(build.IdKey),
-	)
+	version, err := VersionFindOneId(versionId)
+	if err != nil {
+		return errors.Wrapf(err, "getting version '%s'", versionId)
+	}
+	if version == nil {
+		return errors.Wrapf(err, "version '%s' not found", versionId)
+	}
+	if err = version.SetActivated(active); err != nil {
+		return errors.Wrapf(err, "setting activated for version '%s'", versionId)
+	}
+	changed, err := setTaskActivationForVersion(versionId, active, false, nil, caller)
 	if err != nil {
-		return errors.Wrapf(err, "getting builds for version '%s'", versionId)
+		return errors.Wrapf(err, "setting activation for tasks in version '%s'", versionId)
 	}
-	buildIDs := make([]string, 0, len(builds))
-	for _, build := range builds {
-		buildIDs = append(buildIDs, build.Id)
+	if active && changed && evergreen.IsFinishedVersionStatus(version.Status) {
+		if err = version.UpdateStatus(evergreen.VersionCreated); err != nil {
+			return errors.Wrapf(err, "setting status for version '%s'", versionId)
+		}
 	}
+	return nil
+}
 
-	// Update activation for all builds before updating their tasks so the version won't spend
-	// time in an intermediate state where only some builds are updated
-	if err = build.UpdateActivation(buildIDs, active, caller); err != nil {
-		return errors.Wrapf(err, "setting activation for builds in version '%s'", versionId)
+func setTaskActivationForVersion(versionId string, active, withDependencies bool, ignoreTasks []string, caller string) (bool, error) {
```

Comment:
Can we addd a comment to be clear about the boolean being returned?

Reply:
done